rocAL/rocAL/source/coco_meta_data_reader.cpp

```diff
@@ -83,105 +83,210 @@ void COCOMetaDataReader::print_map_contents()
+        new char[file_size + 1],
+        [](char *data)
+        { delete[] data; });
+    f.seekg(0, std::ios::beg);
+    buff.get()[file_size] = '\0';
+    f.read(buff.get(), file_size);
+    f.close();
 
-    _coco_metadata_read_time.start();// Debug timing
-    std::string annotation_file = path;
-    std::ifstream fin;
-    fin.open(annotation_file, std::ios::in);
+    LookaheadParser parser(buff.get());
 
-    std::string str;
-    str.assign(std::istreambuf_iterator<char>(fin), std::istreambuf_iterator<char>());
     BoundingBoxCords bb_coords;
-      BoundingBoxLabels bb_labels;
-      ImgSizes img_sizes;
-
-    Json::Reader reader;
-    Json::Value root;
-    if (reader.parse(str, root) == false) {
-          WRN("Failed to parse Json: " + reader.getFormattedErrorMessages());
-    }
-
-    Json::Value annotation = root["annotations"];
-    Json::Value image = root["images"];
+    BoundingBoxLabels bb_labels;
+    ImgSizes img_sizes;
 
     BoundingBoxCord box;
     ImgSize img_size;
-    
-    for (auto iterator = image.begin(); iterator != image.end(); iterator++)
+    RAPIDJSON_ASSERT(parser.PeekType() == kObjectType);
+    parser.EnterObject();
+    while (const char *key = parser.NextObjectKey())
     {
-        // std::map<int, int,int> id_img_sizes;
-        img_size.h = (*iterator)["height"].asInt();
-        img_size.w = (*iterator)["width"].asInt();
-        img_sizes.push_back(img_size);
-        string image_name  = (*iterator)["file_name"].asString();
-        
-        _map_img_sizes.insert(pair<std::string,std::vector<ImgSize> >(image_name, img_sizes));
-        img_sizes.clear();
+        if (0 == std::strcmp(key, "images"))
+        {
+            RAPIDJSON_ASSERT(parser.PeekType() == kArrayType);
+            parser.EnterArray();
+            while (parser.NextArrayValue())
+            {
+                string image_name;
+                if (parser.PeekType() != kObjectType)
+                {
+                    continue;
+                }
+                parser.EnterObject();
+                while (const char *internal_key = parser.NextObjectKey())
+                {
+                    if (0 == std::strcmp(internal_key, "width"))
+                    {
+                        img_size.w = parser.GetInt();
+                    }
+                    else if (0 == std::strcmp(internal_key, "height"))
+                    {
+                        img_size.h = parser.GetInt();
+                    }
+                    else if (0 == std::strcmp(internal_key, "file_name"))
+                    {
+                        image_name = parser.GetString();
+                    }
+                    else
+                    {
+                        parser.SkipValue();
+                    }
+                }
+                img_sizes.push_back(img_size);
+                _map_img_sizes.insert(pair<std::string, std::vector<ImgSize>>(image_name, img_sizes));
```

Comment:
Why the structure takes a vector of img_sizes? Is there more than one img_size for each image_name?

Reply:
Each image has a different original image & height